Client/RestClient.php

```diff
@@ -165,7 +166,7 @@ public function download(RestRequest $request)
                 if ($e->getPrevious() && $e->getPrevious() instanceof UserException) {
                     throw $e->getPrevious();
                 } else {
-                    throw $e;
+                    throw new UserException($e->getMessage(), $e->getCode(), $e);
```

Comment:
Potrebujeme zachytavat `TooManyRedirects` exception, a previest ju na `UserError`, kedze to nie je chyba extractora.

Tu sa rovno odchytava a prevadza `RequestException`, kedze su tam aj dalsie chyby, ktore ma zmysel previest, aby zbytocne nechodili do logov.

![image](https://user-images.githubusercontent.com/19371734/102196650-7a370080-3ec0-11eb-93b7-1d342db1f512.png)

Chcel som na to napisat test. No zistil som, ze to v Guzzle `5.3` nejde, kedze ten `Mock` handler, sa registruje, pred spracovanie redirectov (`onBefore` event, a redirecty su `onComplete`) ... da sa to az v novej verzii Guzzle. Mozem skusit aktualizovat v samostatnom PR.


Takto sa to testuje v Guzzle 5.3:
https://docs.guzzlephp.org/en/5.3/testing.html

V novej verzii sa to riesi uz inak, a da sa tam otestovat aj redirect:
https://docs.guzzlephp.org/en/stable/testing.html

Reply:
Ta aktualizace guzzle by tomu uricte prospela, ale bude to docela kovbojka, tam je totiz nejakej zadrb s retry https://github.com/keboola/juicer/blob/7fa55e9f190738e69379338ca94171855ee18157/composer.json#L19 
(ale uz je to tak dlouho, ze jsem zapomnel o co slo). Tak urcite samostatne.

Reply:
no to retry je uz cele deprecated: https://github.com/guzzle/retry-subscriber

Ale `RetryMiddleware` je uz priamo sucastou:
https://github.com/guzzle/guzzle/blob/master/src/RetryMiddleware.php

Reply:
vim, ale tam se prave spolihalo na nejake divny chovani neceho a proto je to to zafixovany na ten konkretni commit